pylint/extensions/private_import.py

```diff
@@ -108,7 +108,7 @@ def _name_is_private(name: str) -> bool:
         return (
             bool(name)
             and name[0] == "_"
-            and (len(name) <= 4 or name[1] != "_" or name[-2:] != "__")
+            and not (len(name) > 4 and name[:2] == "__" and name[-2:] == "__")
```

Comment:
The condition was done with great care about performance here, I'm not sure changing the order without benchmark is advisable.

Reply:
I looked at it too fast and thought that the prior check for first char is _ was removed, but it's not. We already checked the first character so name[:2] == "__" should stay name[1] == "_". And in this case it become equivalent.